helix-term/src/ui/picker.rs

```diff
@@ -331,27 +335,35 @@ impl<T: Item> Picker<T> {
             completion_height: 0,
         };
 
-        // scoring on empty input:
-        // TODO: just reuse score()
-        picker.matches.extend(
-            picker
-                .options
-                .iter()
-                .enumerate()
-                .map(|(index, _option)| (index, 0)),
-        );
-
+        picker.re_score(None, true, true);
         picker
     }
 
-    pub fn score(&mut self) {
-        let now = Instant::now();
+    pub fn re_score(
+        &mut self,
+        pattern: Option<String>,
+        reset_cursor: bool,
+        force_recalculation: bool,
+    ) {
+        let cursor = self.cursor;
+        if reset_cursor {
+            self.cursor = 0
+        }
 
-        let pattern = self.prompt.line();
+        let pattern = match pattern {
+            Some(pattern) if pattern == self.previous_pattern && !force_recalculation => return,
+            None if !force_recalculation => return,
+            None => self.previous_pattern.clone(),
+            Some(pattern) => pattern,
+        };
```

Comment:
Would it be better to remove `force_recalculation` in this function and only call this function if it needs to force recalculation?

Reply:
Yeah that might probably be better, I'll tackle this soon...

Reply:
I've replaced it with a dirty flag, is this better?

Reply:
The previous one seemed better to me, can keep it in the argument I guess, don't need to track states.